src/BuildScriptGenerator/CheckerAttribute.cs

```diff
@@ -0,0 +1,19 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Oryx.BuildScriptGenerator
+{
+    [AttributeUsage(AttributeTargets.Class)]
+    public class CheckerAttribute : Attribute
+    {
+        private readonly string[] _targetToolNames;
+
+        public CheckerAttribute(params string[] toolNames) => _targetToolNames = toolNames;
```

Comment:
nit: Change the variable & property to something like `RestrictToTools` just to be clear?